### ISSUE TITLE
M-? global binding does not work on ERB files

### ISSUE THREAD

**Issue by masolino** [purcell/emacs.d#550]
Hi, I'm using your config for edit some Ruby on Rails projects and I
use everywhere your global binding of M-? to search and jump from/to
method definitions.

This command works in all Ruby files but when I use it on a method in
a ERB HTML file it causes the error shown in the screenshot.

I'm on OS X 10.13.3 with Emacs 25.3.

Thanks.
<img width="688" alt="emacs" src="https://user-images.githubusercontent.com/18376/37154290-c9d41a42-22df-11e8-8c94-7f9cb56151dd.png">

**Comment by masolino** [comment]
Ok, I understand that I have to add a keybinding for `sanityinc/counsel-search-project` for the ERB-HTML mode because `M-?` in ERB-HTML mode is the keybinding for `tagedit-convolute-tags`

```
sanityinc/counsel-search-project is an interactive Lisp closure in
‘init-ivy.el’.

(sanityinc/counsel-search-project INITIAL-INPUT &optional
USE-CURRENT-DIR)

Search using ‘counsel-ag’ from the project root for INITIAL-INPUT.
If there is no project root, or if the prefix argument
USE-CURRENT-DIR is set, then search from the current directory
instead.
```

**Comment by dcorking** [comment]
Yes! see for example https://github.com/purcell/emacs.d/blob/661c4feacba8643509f6516fc8084c902fedf807/lisp/init-haskell.el#L13

**Comment by masolino** [comment]
I've ended up using this on my `init-local.el`

```
(require 'tagedit)
(define-key tagedit-mode-map (kbd "M-?") nil)
```

Thank you @dcorking

**Comment by dcorking** [comment]
Thank _you_: I didn't know about `sanityinc/counsel-search-project` until you asked the question, so I installed [ripgrep](https://github.com/BurntSushi/ripgrep) this morning. 🚅

**Comment by purcell** [comment]
That clash with `tagedit` has annoyed me for a while too, so I've just fixed it in master.

`M-?` with `ag` (and more recently `rg`) has transformed the way I work in Emacs. :-)

**Comment by masolino** [comment]
Hi, @purcell, I've pulled your modifications in master and tried to use `M-?` on the same file of my example below but it fail with the same error.
I've also installed `rg`.
Maybe I'm missing something?

**Comment by purcell** [comment]
Hah! You're right - `tagedit` does weird things with its keybindings. Should be fixed properly now in 815ceec. Thanks!

**Comment by masolino** [comment]
Works perfectly now. Thanks
